lib/evmone/analysis.cpp

```diff
@@ -15,6 +16,20 @@ bool is_terminator(uint8_t c) noexcept
     return c == OP_JUMP || c == OP_JUMPI || c == OP_STOP || c == OP_RETURN || c == OP_REVERT ||
            c == OP_SELFDESTRUCT;
 }
+
+std::pair<int, size_t> load_1byte_immediate_value(
+    const uint8_t* code, size_t code_size, size_t index) noexcept
+{
+    return {(index + 1 < code_size) ? code[++index] : 0, index};
```

Comment:
This doesn't look right, because the order of evaluation of `pair` constructor arguments is not defined, is it?

Reply:
I thought this is now specified in C++17, but it is not. To be fixed.

Reply:
(re the need to update `index` conditionally: even if would update it unconditionally on the  calling side and go beyond the code size, it will correctly break out of the loop, so updating here doesn't seem necessary to me)

Reply:
I though about this, but sounds risky. It would not work if we used `!=` in the loop condition, or used an iterator instead of an index. Quite fragile to have the correctness of the implementation of this helper depend on some other loop implementation outside.